torch/linalg/__init__.py

```diff
@@ -8,6 +8,51 @@
 # Note: This not only adds doc strings for functions in the linalg namespace, but
 # also connects the torch.linalg Python namespace to the torch._C._linalg builtins.
 
+cholesky = _add_docstr(_linalg.linalg_cholesky, r"""
+linalg.cholesky(input) -> Tensor
+
+Returns the Cholesky decomposition.
+
+Computes the Cholesky decomposition of a Hermitian (or symmetric for real-valued matrices)
+positive-definite matrix :math:`A` or for batches of Hermitian positive-definite matrices.
+The returned matrix ``L`` is lower-triangular, and
+the decomposition has the form:
+
+.. math::
+
+    A = LL^H
+
+If :attr:`input` is a batch of Hermitian positive-definite
+matrices, then the returned tensor will be composed of lower-triangular Cholesky factors
+of each of the individual matrices.
+
+.. note:: If the :attr:`input` is not Hermitian positive-definite matrix a RuntimeError is raised
```

Comment:
This note also needs to account for the batching behavior. Maybe something like:

"If :attr:\`input\` is not a Hermitian positive-definite matrix, or if it's a batch of matrices and one of them is not a Hermitian positive-definite matrix, then a RuntimeError will be thrown."

As for the error behavior: does it specify which matrix in the batch was discovered to not be a Hermitian positive-definite matrix, too? Another option here would be to not elaborate about the content of the error message.

Reply:
Batch is also specified in the error message.
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebraUtils.h#L104

Reply:
Great; let's be clear about that (and maybe even add a test for it erroring on the correct batch element).